src/OpenFOAM/meshes/primitiveMesh/PrimitivePatch/PrimitivePatch.H

```diff
@@ -179,6 +179,7 @@ private:
 
         //- Face centres
         mutable Field<PointType>* faceCentresPtr_;
+        mutable gpuField<PointType> gpuFaceCentresPtr_;
```

Comment:
I think that we do not have to allocate additional memory for face centers since this can be calculated using the center method in faceData. Also I would like to reduce memory usage wherever possible.